docs/reference/executionModes.md

```diff
@@ -3,145 +3,277 @@ id: executionModes
 
-This execution mode has a performance drawback as it has to query the maximum value for *compareCol* on input and output in each SDLB job.
+Its rhythm can be configured by setting a `triggerType` to either `Once`, `ProcessingTime` or `Continuous` (default is `Once`).
+When using `ProcessingTime` or `Continuous` you can configure the interval through the attribute `triggerTime`.
 
-### DataObjectStateIncrementalMode
-Performance can be optimized by remembering the state of the last increment. The state is saved in a .json file for which the user needs to provide a path using the `--state-path` option, see [Command Line](commandLine.md).
+:::info
+You need to start your SDLB run with the `--streaming` flag to enable streaming mode.
+If you don't, SDLB will always use `Once` as trigger type (single microbatch) and the action is executed synchronously.
+:::
+
+Spark Structured Streaming is keeping state information about processed data.
+To do so, it needs a `checkpointLocation` configured on the SparkStreamingMode.
+
+Note that *Spark Structured Streaming* needs an input DataObject that supports the creation of streaming DataFrames.
+For the time being, only the input sources delivered with Spark Streaming are supported.
+These are `KafkaTopicDataObject` and all `SparkFileDataObjects`, see also [Spark StructuredStreaming](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#creating-streaming-dataframes-and-streaming-datasets).
+
+
+### KafkaStateIncrementalMode
+A special incremental execution mode for Kafka inputs, remembering the state from the last increment through the Kafka Consumer, 
+e.g. committed offsets.
 
-*DataObjectStateIncrementalMode* needs a main input DataObject that can create incremental output, e.g. JdbcTableDataObject, SparkFileDataObject or AirbyteDataObject. The inner meaning of the state is defined by the DataObject and not interpreted outside of it. Some DataObjects have a configuration option to define the incremental output, e.g. JdbcTableDataObject.incrementalOutputExpr, others just use technical timestamps existing by design, e.g. SparkFileDataObject. 
 
 ### FileIncrementalMoveMode
-Another paradigm for incremental processing with files is to move or delete input files once they are processed. This can be achieved by using FileIncrementalMoveMode. If option *archiveSubdirectory* is configured, files are moved into that directory after processing, otherwise they are deleted.
+Another paradigm for incremental processing with files is to move or delete input files once they are processed.
+This can be achieved by using FileIncrementalMoveMode. If option `archiveSubdirectory` is configured, files are moved into that directory after processing, otherwise they are deleted.
 
 FileIncrementalMoveMode is the only execution mode that can be used with the file engine (see also [Execution engines](executionEngines.md)), but also with SparkFileDataObjects and the data frame engine.
```

Comment:
I dont think it "is the only" one, PartitionDiffMode can also be used with file and data frame engine.
```suggestion
FileIncrementalMoveMode can be used with the file engine (see also [Execution engines](executionEngines.md)), but also with SparkFileDataObjects and the data frame engine.
```

Reply:
Two small comments, otherwise ok to merge :-)